### ISSUE TITLE
Tests: Possible search context leak in DeleteByQueryTests

### ISSUE THREAD

**Issue by imotov** [elastic/elasticsearch#13580]
See examples here: 
http://build-us-00.elastic.co/job/es_core_master_centos/7546/
http://build-us-00.elastic.co/job/es_g1gc_master_metal/17398/
http://build-us-00.elastic.co/job/es_core_master_window-2012/1735/

The tests fail in [assertSearchContextsClosed](https://github.com/elastic/elasticsearch/blob/master/plugins/delete-by-query/src/test/java/org/elasticsearch/plugin/deletebyquery/DeleteByQueryTests.java#L446) with the error:

```
FAILURE 0.06s J2 | DeleteByQueryTests.testDeleteByQueryWithNoIndices <<<
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <0L>
   >      but: was <1L>
   > 	at __randomizedtesting.SeedInfo.seed([1B261257DA113457:444F85E3E99286BE]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.plugin.deletebyquery.DeleteByQueryTests.assertSearchContextsClosed(DeleteByQueryTests.java:446)
   > 	at org.elasticsearch.plugin.deletebyquery.DeleteByQueryTests.testDeleteByQueryWithNoIndices(DeleteByQueryTests.java:73)
   > 	at java.lang.Thread.run(Thread.java:745)
```

Once a test fails all tests after it would fail as well. I wasn't able to reproduce the tests with the seed that it was failing with in CI, but I was able to reproduce it once by running the entire suite many times with random seeds. Unfortunately, after it happened it doesn't fail on my machine anymore.

**Comment by rjernst** [comment]
Another example:
http://build-us-00.elastic.co/job/es_g1gc_master_metal/17555/

**Comment by tlrx** [comment]
Test bug: the scroll id (and thus the underlying search context) used in DeleteByQuery is cleared in an async manner at the end of the scrolling in TransportDeleteByQueryAction. So the test should wait a bit for the search context to be released.

**Comment by areek** [comment]
Another instance of this failure: http://build-us-00.elastic.co/job/es_core_2x_regression/118/, seems like it was back ported to 2.x as well

**Comment by javanna** [comment]
I am reopening this, the test still fails at times, I couldn't repro yet though unfortunately: 
http://build-us-00.elastic.co/job/es_core_master_window-2008/2386/
http://build-us-00.elastic.co/job/es_feature_ingest/194/

**Comment by s1monw** [comment]
this is caused by a stats bug fixed in #13801

**Comment by s1monw** [comment]
fixed by #13801
